api/projects/projects-model.js

```diff
@@ -0,0 +1,66 @@
+const knex = require('knex')
+
+const config = require('../../knexfile')
+
+const db = knex(config.development)
+
+module.exports = {
+    add,
+    attachResource,
+    get,
+    getTree
+}
+
+function add(item) {
+    const {name, description, completed} = item
+    return db('projects').insert({name, description, completed: !!completed}).then(([id]) => get(id))
+}
+
+function get(id) {
+    if (id) {
+        return db('projects')
+        .where({id}).first()
+        .then(resp => {
+            resp.completed = !!resp.completed
+            return resp

```

Comment:
If there is no project returned, then `resp.completed` will throw an error. Adding a `if (resp)` should fix this error '' '